rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -2110,6 +2110,148 @@ private static Stream<BridgeConstants> provideBridgeConstants() {
         return Stream.of(BridgeRegTestConstants.getInstance(), BridgeTestNetConstants.getInstance(), BridgeMainNetConstants.getInstance());
     }
 
+    @Test
+    void rskTxWaitingForSignature_fail_adding_an_already_existence_key_after_rskip_375() throws IOException {
+        // Arrange
+        BridgeConstants bridgeConstants = BridgeRegTestConstants.getInstance();
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP146)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP176)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP271)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP375)).thenReturn(true);
+
+        // Set state to make concur a pegout migration tx and pegout batch creation on the same updateCollection
+        Federation oldFederation = bridgeConstants.getGenesisFederation();
+        Federation newFederation = new Federation(
+            FederationTestUtils.getFederationMembers(1),
+            Instant.EPOCH,
+            5L,
+            bridgeConstants.getBtcParams()
+        );
+
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+
+        Set<PegoutsWaitingForConfirmations.Entry> pegoutsWaitingForConfirmationsEntries = new HashSet<>();
+        PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations = new PegoutsWaitingForConfirmations(pegoutsWaitingForConfirmationsEntries);
+        when(provider.getPegoutsWaitingForSignatures()).thenReturn(new TreeMap<>());
+
+        when(provider.getFeePerKb())
+            .thenReturn(Coin.MILLICOIN);
+        when(provider.getReleaseRequestQueue())
+            .thenReturn(new ReleaseRequestQueue(PegTestUtils.createReleaseRequestQueueEntries(3)));
+        when(provider.getPegoutsWaitingForConfirmations())
+            .thenReturn(pegoutsWaitingForConfirmations);
+
+        // Federation change on going
+        when(provider.getOldFederation())
+            .thenReturn(oldFederation);
+        when(provider.getNewFederation())
+            .thenReturn(newFederation);
+
+        // Utxos to migrate
+        List<UTXO> utxos = PegTestUtils.createUTXOs(30, oldFederation.getAddress());
+        when(provider.getOldFederationBtcUTXOs())
+            .thenReturn(utxos);
+
+        List<UTXO> utxosNew = PegTestUtils.createUTXOs(25, newFederation.getAddress());
+        when(provider.getNewFederationBtcUTXOs())
+            .thenReturn(utxosNew);
+
+        // Advance blockchain to migration phase
+        BlockGenerator blockGenerator = new BlockGenerator();
+        org.ethereum.core.Block rskCurrentBlock = blockGenerator.createBlock(180, 1);
+        BridgeEventLogger bridgeEventLogger = mock(BridgeEventLogger.class);
```

Comment:
Do you need this? BridgeSupportBuilder will create a mock by default

Reply:
Solved